sources/iTermNotificationController.m

```diff
@@ -84,6 +85,53 @@ - (BOOL)notify:(NSString *)title
                       sticky:NO];
 }
 
+- (BOOL)notifyRich:(NSString *)title
+      withSubtitle:(NSString *)subtitle
+   withDescription:(NSString *)description
+         withImage:(NSString *)image
+         withSound:(NSString *)sound
+       windowIndex:(int)windowIndex
+          tabIndex:(int)tabIndex
+         viewIndex:(int)viewIndex {
+    DLog(@"Notify title=%@ description=%@ window=%@ tab=%@ view=%@",
+            title, description, @(windowIndex), @(tabIndex), @(viewIndex));
+
+    if (description == nil) {
+        DLog(@"notifyRich passed nil description.");
+        return NO;
+    }
+
+    // Send to NSUserDefaults directly
+    NSUserNotification *notification = [[NSUserNotification alloc] init];
+    notification.informativeText = description;
+
+    NSDictionary *context = nil;
+    if (windowIndex >= 0) {
+        context = @{ @"win": @(windowIndex),
+                @"tab": @(tabIndex),
+                @"view": @(viewIndex) };
+    }
+    notification.userInfo = context;
+
+    if (title != nil) {
+        notification.title = title;
+    }
+    if (subtitle != nil) {
+        notification.subtitle = subtitle;
+    }
+    if (image != nil) {
+        NSData *data = [NSData dataWithContentsOfFile:image];
+        notification.contentImage = [[[iTermImage imageWithCompressedData:data] images] firstObject];
```

Comment:
What if `data` is nil?

Reply:
Ah, I had overlooked that.  It would appear `iTermImage` will handle the `nil` case correctly, but I've added a conditional to avoid the call altogether.